### ISSUE TITLE
entgql: possibility to attach custom directives to mutation create and update inputs

### ISSUE THREAD

**Issue by lucvankessel** [ent/ent#3407]
I want to attach a custom directive to a generated mutation create/update field.

- [x] I have searched the [issues](https://github.com/ent/ent/issues) of this repository and believe that this is not a duplicate.

## Summary 💡
Through the mutations in a field definition in a schema i want to be able to do something like entgql.UpdateMutation.Directive('directive name', {args}) to be able to attach the given directive to it. (using the same syntax as the current way to add directives to the gql type would be preffered)

## Motivation 🔦

Right now i can create a custom directive for checking the minimum length of a string(for example). but this directive is executed whenever i get a object from the database and not when i insert one through a create/update input.
This leads to me being unable to attach directives to places i want them on. manually this would be possible but we generate our create and update inputs through the mutations and want to attach them in the schema/field definitions.

**Comment by lucvankessel** [comment]
See the merge request above for the fix i wanted.
In there i also added a question about adding the skipping of adding directives to certain types/field/mutations.

**Comment by a8m** [comment]
Hey, @lucvankessel 👋 
Would you mind putting here an example of the mutation inputs you want to generate with entgql? `input` directives or field-level directives?

Just to make sure you're not blocked on this, check out my `SchemaHook` example on: https://github.com/ent/ent/issues/3173#issuecomment-1359338499

**Comment by lucvankessel** [comment]
hey @a8m,
This is a simple example of what i want:
```
input CreateProjectInput {
  name: String! @CustomDirective("some arg")
}
```
This input is then generated from the mutations in the schema.
I can generate the inputs without the directive.
[This MR](https://github.com/ent/contrib/pull/469) does what i want it to do.
In the future it might also be nice to have the ability to skip these directives on either the type, create or update inputs.

**Comment by lucvankessel** [comment]
Hey @a8m,
Can i help you guys in any way to fast track the progress on this issue?

**Comment by lucvankessel** [comment]
Made this [MR]{https://github.com/ent/contrib/pull/543} a few days ago with a proposed backwards compatible solution. let me know what you think and if it needs changing etc.
